src/category.rs

```diff
@@ -1,57 +1,244 @@
+                                      }
+                                      element
+                                  }))?;
 
-        try!(writer.write(Event::Text(Element::new(self.name.as_str()))));
+        writer.write(Event::Text(Element::new(self.name
+                                                  .as_str())))?;
 
         writer.write(Event::End(element))
     }
 }
+
+/// This `CategoryBuilder` struct creates the `Category`.
+#[derive(Debug, Clone, Default)]
+pub struct CategoryBuilder
+{
+    name: String,
+    domain: Option<String>,
+}
+
+impl CategoryBuilder
+{
+    /// Construct a new `CategoryBuilder` and return default values.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use rss::CategoryBuilder;
+    ///
+    /// let category_builder = CategoryBuilder::new();
+    /// ```
+    pub fn new() -> CategoryBuilder
+    {
+        CategoryBuilder::default()
+    }
+
+    /// Set the category that exists under `Category`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use rss::CategoryBuilder;
+    ///
+    /// let mut category_builder = CategoryBuilder::new();
+    /// category_builder.name("Podcast");
+    /// ```
+    pub fn name(&mut self,
+                name: &str)
+        -> &mut CategoryBuilder
+    {
+        self.name = name.to_owned();
+        self
+    }
+
+    /// Set the optional domain that exists under `Category`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use rss::CategoryBuilder;
+    ///
+    /// let mut category_builder = CategoryBuilder::new();
+    /// category_builder.domain(Some("http://www.example.com".to_owned()));
+    /// ```
+    pub fn domain(&mut self,
+                  domain: Option<String>)
+        -> &mut CategoryBuilder
+    {
+        self.domain = domain;
+        self
+    }
+
+    /// Validate the contents of `Category`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use rss::CategoryBuilder;
+    ///
+    /// let mut category_builder = CategoryBuilder::new();
+    /// category_builder.domain(Some("http://www.example.com".to_owned()));
+    /// category_builder.name("Podcast");
+    /// category_builder.validate().unwrap();
+    /// category_builder.finalize().unwrap();
+    /// ```
+
+    pub fn validate(&mut self) -> Result<&mut CategoryBuilder, Error>
+    {
+        let domain = self.domain
+                         .clone();
+        if domain.is_some() {
+            Url::parse(domain.unwrap()
+                             .as_str())?;
+        }
+
+        Ok(self)
+    }
+
+    /// Construct the `Category` from the `CategoryBuilder`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use rss::CategoryBuilder;
+    ///
+    /// let category = CategoryBuilder::new()
+    ///         .name("Title")
+    ///         .domain(None)
+    ///         .finalize()
+    ///         .unwrap();
+    /// ```
+    pub fn finalize(&self) -> Result<Category, Error>
```

Comment:
To remove the clones you could have each of the builder methods consume self and return self instead of taking a mutable reference.

https://aturon.github.io/ownership/builders.html#consuming-builders: